File: app/src/main/java/id/coedotz/appbiodata/adapter/AdapterData.java
package id.coedotz.appbiodata.adapter;

import android.content.Context;
import android.content.Intent;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import id.coedotz.appbiodata.MainActivity;
import id.coedotz.appbiodata.model.DataModel;
import id.coedotz.appbiodata.*;

import java.util.List;

public class AdapterData extends RecyclerView.Adapter<AdapterData.HolderData> {

    private List<DataModel> mList ;
    private Context ctx;


    public  AdapterData (Context ctx, List<DataModel> mList)
    {
        this.ctx = ctx;
        this.mList = mList;
    }

    @Override
    public HolderData onCreateViewHolder(ViewGroup parent, int viewType) {
        View layout = LayoutInflater.from(parent.getContext()).inflate(R.layout.layoutlist,parent, false);
        HolderData holder = new HolderData(layout);
        return holder;
    }

    @Override
    public void onBindViewHolder(HolderData holder, int position) {
        DataModel dm = mList.get(position);
        holder.nama.setText(dm.getNama());
        holder.usia.setText(dm.getUsia());
        holder.domisili.setText(dm.getDomisili());
        holder.dm = dm;
    }

    @Override
    public int getItemCount() {
        return mList.size();
    }


    class HolderData extends  RecyclerView.ViewHolder{
        TextView nama, domisili, usia;
        DataModel dm;
        public HolderData (View v)
        {
            super(v);

            nama  = (TextView) v.findViewById(R.id.tvNama);
            usia = (TextView) v.findViewById(R.id.tvUsia);
            domisili = (TextView) v.findViewById(R.id.tvDomisili);

            v.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    Intent goInput = new Intent(ctx, MainActivity.class);
                    goInput.putExtra("id", dm.getId());
                    goInput.putExtra("nama", dm.getNama());
                    goInput.putExtra("usia", dm.getUsia());
                    goInput.putExtra("domisili", dm.getDomisili());

                    ctx.startActivity(goInput);
                }
            });
        }
    }
}

File: app/src/main/java/id/coedotz/appbiodata/adapter/AdapterData.kt
package id.coedotz.appbiodata.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import id.coedotz.appbiodata.MainActivity
import id.coedotz.appbiodata.R
import id.coedotz.appbiodata.model.DataModel

class AdapterData(private val ctx: Context, private val mList: List<DataModel>) : RecyclerView.Adapter<AdapterData.HolderData>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): HolderData {
        val layout = LayoutInflater.from(parent.context).inflate(R.layout.layoutlist, parent, false)
        return HolderData(layout)
    }

    override fun onBindViewHolder(holder: HolderData, position: Int) {
        val dm = mList[position]
        holder.nama.text = dm.nama
        holder.usia.text = dm.usia
        holder.domisili.text = dm.domisili
        holder.dm = dm
    }

    override fun getItemCount(): Int {
        return mList.size
    }

    inner class HolderData(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var nama: TextView = itemView.findViewById(R.id.tvNama)
        var usia: TextView = itemView.findViewById(R.id.tvUsia)
        var domisili: TextView = itemView.findViewById(R.id.tvDomisili)
        var dm: DataModel? = null

        init {
            itemView.setOnClickListener {
                val goInput = Intent(ctx, MainActivity::class.java)
                goInput.putExtra("id", dm?.id)
                goInput.putExtra("nama", dm?.nama)
                goInput.putExtra("usia", dm?.usia)
                goInput.putExtra("domisili", dm?.domisili)
                ctx.startActivity(goInput)
            }
        }
    }
}